### PR TITLE
feat: add server routes

### PR DESCRIPTION
**Pull request by karansapolia**
add routes for auth, user, timeline, categories

fix: #28

( To setup routes quick for everyone to start work, will push routes tests in next PR )

## PR Checklist
Please check that your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/pesto-students/batch-10-tripen/wiki/Commit-message-guidelines)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] GIFs/Videos/Images added (compulsory for frontend changes)

## PR Type
What kind of change does this PR introduce? 

- [ ] Bugfix
- [x] Feature
- [ ] Tests changed / added / updated
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Infrastructure changes
- [ ] Dependency upgrade
- [ ] Other... Please describe:

## Issue Number: 
#28

## What is the new behavior?
Routes for auth, user, timeline and categories functioning

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: server/src/routes/api/v1/auth.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+router.post('/signin', [/*signinValidation*/], /*signinController*/ (req, res) => {
+  res.status(202).json({ 'status': 'signin controller reached'}); 
+}); // validations & controllers TODO
+router.post('/signup', [/*signupValidation*/], /*signupController*/ (req, res) => {
+  res.status(201).json({ 'status': 'signup controller reached'});
+}); // validations & controllers TODO
+router.post('/forgot-password', [/*forgotpasswordValidation*/], /*forgotpasswordController*/ (req, res) => {
+  res.status(200).json({ 'status': 'forgot password controller reached' });
+}); // validations & controllers TODO
+
```
Comment (p-racoon) [inline server/src/routes/api/v1/auth.js]: Lots of comments

File: server/src/routes/api/v1/index.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import timelineRoutes from './timeline';
+import categoriesRoutes from './categories';
+
+import { Router } from 'express';
+
+const router = Router();
+router.use('/auth', authRoutes);
+router.use('/user', userRoutes);
+router.use('/timeline', timelineRoutes);
+router.use('/categories', categoriesRoutes);
```
Comment (p-racoon) [inline server/src/routes/api/v1/index.js]: Line Gaps

File: server/src/routes/api/v1/auth.js (hunk 1 above)
Comment (karansapolia) [reply server/src/routes/api/v1/auth.js]: yeah, the comments are middlewares and controllers. Portions @thapasya-m and @ArpitKumar2092 have worked on. So these are just placeholders for now.

File: server/src/routes/api/v1/index.js (hunk 2 above)
Comment (karansapolia) [reply server/src/routes/api/v1/index.js]: [Done as requested](https://github.com/pesto-students/batch-10-tripen/blob/6643af3ccf9dd01db48f90da6c259d2ff13c6c51/server/src/routes/api/v1/index.js#L7) :+1:

**Pull request merged**